src/Utilities/cachingoptimizer.jl

```diff
@@ -80,11 +80,13 @@ typeof(model_cache)}` so it does not support the function
 """
 function CachingOptimizer(
     model_cache::MOI.ModelLike,
-    optimizer::MOI.AbstractOptimizer,
+    optimizer::MOI.AbstractOptimizer;
+    auto_bridge::Bool = false,
 )
     @assert MOI.is_empty(model_cache)
     @assert MOI.is_empty(optimizer)
-    return CachingOptimizer{typeof(optimizer),typeof(model_cache)}(
+    T = auto_bridge ? MOI.AbstractOptimizer : typeof(optimizer)
```

Comment:
Having a non-concrete type for the optimizer might be quite bad for performance. As JuMP already has a non-concrete type, maybe we should change the type of the backend instead. Another option is to use a small union of the two possibilities here.

Reply:
Removed.